### PR TITLE
Reduced amount of Docker Image layers

### PR DESCRIPTION
**Pull request by madduci**
:warning: This is a Pull Request Template :warning:

> Checkout all the things you've done by placing an `x` within the square braces below.

## Proposed Changes
- [X] I've read the [CODE OF CONDUCT](https://github.com/my-first-pr/hacktoberfest-2018/blob/master/CODE_OF_CONDUCT.md) and abide to it.
- [X] I've read the [CONTRIBUTING.md](https://github.com/my-first-pr/hacktoberfest-2018/blob/master/CONTRIBUTING.md)
- [X] I've forked the repository.
- [X] I've created a branch and made my changes in it.
- [X] I understand opening a PULL REQUEST doesn't mean it will be merged for sure.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -79,7 +73,9 @@ RUN set -ex
 #
 # (!!!) Turns off SSL verification on the whole system (!!!)
 #
-RUN if [ "$NO_SSL" = "True" ]; then \
+# https://www.peterbe.com/plog/set-ex
+RUN set -ex \
```
Comment (blockspacer) [inline Dockerfile:77]: docker prints error without extra `&& \`
```
RUN set -ex \
  && \
  if
```

**Review by blockspacer — COMMENTED** [review]
dockerfile changed a lot in recent master, needs update

**Comment by blockspacer** [comment]
Thanks a lot, it is good idea to reduce amount of Docker Image layers

**Comment by blockspacer** [comment]
Note what now Docker build is broken https://github.com/blockspacer/CXTPL/issues/52

**Comment by madduci** [comment]
i'll have a look at it. I have 2 questions:

1. do you actually need all the versions of LLVM installed? That a huge image for sure.

2. all the code relies only on Folly and Gtest for tests,or do you use also other libraries independently? If only folly is your main requirement, bincrafters have a Conan recipe for folly (duckduckgo for 'conan-folly')

**Comment by blockspacer** [comment]
> i'll have a lot at it. I have 2 questions:
> 
> 1. do you actually need Al the versions of LLVM installed? That a huge image for sure.
> 2. all the code relies only on Folly and Gtest for tests,or do you use also other libraries independently? If only folly is your main requirement, bincrafters have a Conan recipe for folly (duckduckgo for 'conan-folly')

1. We need llvm-6 only for Cling or iwyu. Both optional, so i plan to add ARG `enable_llvm` to cmakelists. Also multistage builds can be used for separation of dev-env & release build.

2. I`ll try to add folly from git and use Conan with custom patch, like so https://github.com/studiofuga/conan-proj/blob/master/conanfile.py#L23

**Pull request closed without merging**